### PR TITLE
Use StatusForEventUIDs array  and OrganizationTypes from API

### PR DESCRIPTION
**Pull request by JamesSCBD**

### CODE REVIEW

File: components/list/ContactsList.vue [hunk 1]
```diff
@@ -88,17 +72,33 @@ import mixin from './mixin';
...
+    key: 'OrganizationName', label: 'Organization', sortable: true, ColumnType: 'BaseColumn',
+  },
+  {
+    key: 'Emails', label: 'Email', sortable: true, ColumnType: 'BaseColumn',
+  },
+  {
+    key: 'Country', label: 'Country', sortable: true, class: 'text-left', ColumnType: 'BaseColumn',
+  },
+  {
+    key: 'Score', label: 'Rank', sortable: true, ColumnType: 'BaseColumn',
```
Comment (sbilodeau) [inline components/list/ContactsList.vue:100]: what has chnaged?

File: components/list/ContactsList.vue [hunk 2]
```diff
@@ -214,21 +219,20 @@ function updateColumns() {
...
-        },
-      ];
-    }
+    this.columns = [
+      ...this.columns,
+      {
+        key       : `StatusEvents[${index}].Date`,
+        label     : this.getEventCodeById(eventUIDs[index]),
+        sortable  : true,
+        ColumnType: 'MeetingsColumn',
```
Comment (sbilodeau) [inline components/list/ContactsList.vue:233]: I find this name `MeetingsColumn` wrong. this column is about *RegistrationStatus*, not meeting

File: components/list/ContactsList.vue [hunk 3]
```diff
@@ -261,6 +265,19 @@ function cleanContact(item) {
...
+}
+
+function bindStatusEventDate(eventStatus) {
+  if (eventStatus) {
+    const statusDate =  this.$options.filters.datetime(eventStatus.Date);
+    return eventStatus.IsFunded ? `${statusDate} $` : statusDate;
+  }
+  return '';
+}
+
```
Comment (sbilodeau) [inline components/list/ContactsList.vue:280]: Maybe better let the cell template handle this logic? You can also extract to RegistrationStatus column type.

eg:

RegistrationStatusColumn.vue

```html
<template>
   <div :class="{ [registrationStatus[eventStatus.Status]] :  true  }"> 
    <span>{{ registrationStatus.Date | dateTime }}</span>
    <span v-if="registrationStatus.isFunded">$</span>
</template>

```

File: components/list/ContactsList.vue [hunk 4]
```diff
@@ -55,26 +55,10 @@
...
-        <span
-          :class="registrationStatuses[data.item.StatusEvent3]"
-        >{{ data.item.StatusEvent3Date | datetime }}</span>
-      </template>
-      <template v-slot:cell(StatusEvent4Date)="data">
-        <span
-          :class="registrationStatuses[data.item.StatusEvent4]"
-        >{{ data.item.StatusEvent4Date | datetime }}</span>
+      <template v-for="(column, index) in meetingsColumns" v-slot:[`cell(${column.key})`]="data">
+        <span :key="column.key" :class="setStatusColor(data.item.StatusEvents[index])">
```
Comment (sbilodeau) [inline components/list/ContactsList.vue:59]: `:class` binding normally work with object or array syntax

https://vuejs.org/v2/guide/class-and-style.html

File: components/list/ContactsList.vue [hunk 5]
```diff
@@ -214,21 +219,20 @@ function updateColumns() {
...
-          key     : `StatusEvent${index + 1}Date`,
-          label   : this.getEventCodeById(eventUIDs[index]),
-          sortable: true,
-        },
-      ];
-    }
+    this.columns = [
+      ...this.columns,
+      {
+        key       : `StatusEvents[${index}].Date`,
```
Comment (sbilodeau) [inline components/list/ContactsList.vue:230]: use eventuid as key whould be better.

File: components/list/ContactsList.vue [hunk 6]
```diff
@@ -214,21 +219,20 @@ function updateColumns() {
...
+        label     : this.getEventCodeById(eventUIDs[index]),
+        sortable  : true,
+        ColumnType: 'MeetingsColumn',
+      },
+    ];
```
Comment (sbilodeau) [inline components/list/ContactsList.vue:235]: Your loop is incrementally adding column by re-affect this.columns array as many time as eventUIDs.length. 
it's working but not easy to understand. 

better to create new columns, then concatenate them all at once

eg:

```javascript
const eventUIDs = query.StatusForEventUIDs || [];

const statusColumns = eventUIDs.map(eid=>{
  Key: `statusFor-${eid}`,
  Label: this.getEventCodeById(eid),
  sortable  : true,
 ColumnType: 'MeetingsColumn',
})

this.columns = [...baseColums, ...statusColumns];

```

File: components/list/ContactsList.vue (hunk 1 above)
Comment (JamesSCBD) [reply components/list/ContactsList.vue]: Added ColumnType But now removed

File: components/list/ContactsList.vue (hunk 2 above)
Comment (JamesSCBD) [reply components/list/ContactsList.vue]: Rename with `RegStatusColumn`

File: components/list/ContactsList.vue (hunk 3 above)
Comment (JamesSCBD) [reply components/list/ContactsList.vue]: Done

File: components/list/ContactsList.vue (hunk 4 above)
Comment (JamesSCBD) [reply components/list/ContactsList.vue]: I will make sure next time

File: components/list/ContactsList.vue (hunk 5 above)
Comment (JamesSCBD) [reply components/list/ContactsList.vue]: Done! Use like this `StatusEvents[${index}].EventUID`

File: components/list/ContactsList.vue (hunk 6 above)
Comment (JamesSCBD) [reply components/list/ContactsList.vue]: Done

Comment (JamesSCBD) [reply components/list/ContactsList.vue]: Key: `statusFor-${eid}` is not woking so i use like: `StatusEvents[${index}].EventUID`

**Pull request merged**